web/yo/app/scripts/controllers/gene.js

```diff
@@ -1890,15 +1952,116 @@ angular.module('oncokbApp')
                     }
                     mutationChanged = false;
                 }
+                if(_.isEmpty(myGeneTypeEvidence) && _.isEmpty(myUpdatedEvidences) && _.isEmpty(myDeletedEvidences)) {
+                    $scope.allMyChanges = true;
+                }
             }
-
-            $scope.accept = function(event, type, mutation, tumor, TI, treatment, reviewObj) {
-                if (event !== null) {
-                    $scope.stopCollopse(event);
+            var apiCallCount = 0;
+            function doneSaving() {
+                if(apiCallCount === 1) {
+                    $scope.$emit('doneSaveDataToDatabase');
+                    $scope.allMyChanges = true;
+                    myUpdatedEvidences = {};
+                    myDeletedEvidences = [];
+                    myGeneTypeEvidence = {};
+                    myUpdatedEvidenceModels = [];
+                    myDeletedEvidenceIndice = [];
+                }else {
+                    apiCallCount--;
                 }
-                if (reviewObj.get('action')) {
-                    return;
+            }
+            $scope.acceptAllMyChanges = function() {
+                if ($scope.status.isDesiredGene) {
+                    $scope.$emit('startSaveDataToDatabase');
                 }
+                if(_.isEmpty(myGeneTypeEvidence) && _.isEmpty(myUpdatedEvidences) && !_.isEmpty(myDeletedEvidences)) {
```

Comment:
Probably can be changed to $q.all, see example here http://stackoverflow.com/questions/21310964/angularjs-q-all